File: sms/backend/src/main/java/com/society/management/service/AnnouncementService.java
package com.society.management.service;

import com.society.management.dto.request.AnnouncementRequest;
import com.society.management.dto.response.AnnouncementDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

/**
 * Announcement management operations.
 */
public interface AnnouncementService {

    AnnouncementDto create(Long createdByUserId, AnnouncementRequest request);

    AnnouncementDto update(Long id, AnnouncementRequest request);

    AnnouncementDto getById(Long id);

    Page<AnnouncementDto> getBySociety(Long societyId, Pageable pageable);

    List<AnnouncementDto> getActiveAnnouncements(Long societyId);

    void delete(Long id);

    void sendNotification(Long announcementId);
}

File: Society-Management-System/frontend/src/services/announcementService.js
import api from './api'

const BASE = '/api/v1/announcements'

export const announcementService = {
  getAll: (societyId, params = {}) =>
    api.get(BASE, { params: { societyId, ...params } }),

  getActive: (societyId) =>
    api.get(`${BASE}/active`, { params: { societyId } }),

  getById: (id) =>
    api.get(`${BASE}/${id}`),

  create: (data) =>
    api.post(BASE, data),

  update: (id, data) =>
    api.put(`${BASE}/${id}`, data),

  delete: (id) =>
    api.delete(`${BASE}/${id}`),

  notifyAll: (id) =>
    api.post(`${BASE}/${id}/notify`),
}
